src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1260,50 +1253,44 @@ public void setBedSpawnLocation(Location location, boolean force)
 	@Override
 	public boolean getAllowFlight()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return canFly;
 	}
-	
+	boolean canFly;
+	boolean isFlying;
 	@Override
 	public void setAllowFlight(boolean flight)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		canFly = flight;
 	}
 	
 	@Override
 	public void hidePlayer(Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hiddenPlayers.put(this,player);
 	}
 	
 	@Override
 	public void showPlayer(Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		hiddenPlayers.remove(this,player);
 	}
-	
+	HashMap<Player,Player> hiddenPlayers = new HashMap<>();
 	@Override
 	public boolean canSee(Player player)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return hiddenPlayers.containsKey(this) && hiddenPlayers.containsValue(player);
 	}
```

Comment:
In implementation of this 3 methods there's no need for a HashMap, because every time the key would be just "this", maybe this would make any sense if this hiddenPlayers HashMap would be static, but the code for canSee(Player) method would not work. It would be much more logical to just create a List<Player> that would store hidden players for each player.

Reply:
I'll make a set, because i dont want repeated elements in the list

Reply:
Done